analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.CSharp7.cs

```diff
@@ -39,20 +39,17 @@ void VariableDesignationPattern_Variable(object o)
             if (o is string s)
             {
                 if (s == null) // Noncompliant, always false
-                { // Secondary, not executed code
+                {
                     s.ToString();
                 }
             }
         }
 
         void VariableDesignationPattern_Source(object o)
         {
-            // We can set NotNull constraint only for one of the variables in the if condition
-            // and we choose the declared variable because it is more likely to have usages of
-            // it inside the statement body.
             if (o is string s)
             {
-                if (o == null) // Compliant, False Negative
+                if (o == null)  // Noncompliant: always null
```

Comment:
I'm not sure I understand why this is noncompliant - I don't get the point of the example.
Maybe it should have been `Noncompliant: always false`?

Reply:
The comment was wrong.